### PR TITLE
Routing on graphs

### PR DESCRIPTION
**Pull request by friedjoff**
Support routing on multiple graphs.

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] Labels applied. if it's a release? a hotfix?
- [x] Tests added.

### CODE REVIEW

File: src/ol/controls/RoutingControl.js [hunk 1]
```diff
@@ -63,6 +63,9 @@ class RoutingControl extends Control {
       },
     });
 
+    /** @ignore */
+    this.graphs = options.graphs || [];
```
Comment (oterral) [inline src/ol/controls/RoutingControl.js:67]: Could we integrate the 'osm' graph value by default in the default array ?

File: src/ol/controls/RoutingControl.js [hunk 2]
```diff
@@ -215,6 +218,63 @@ class RoutingControl extends Control {
...
+    // Create point features for the viaPoints
+    this.viaPoints.forEach((viaPoint, idx) => {
+      return this.drawViaPoint(viaPoint, idx);
+    });
+
+    this.graphs
+      .filter(([graph]) => graph !== 'osm')
+      .forEach(([graph], index) =>
+        this.api
+          .route(
```
Comment (oterral) [inline src/ol/controls/RoutingControl.js:233]: to avoid duplication of code . Could we track all the promises then return  Promise.all() ?

**Review by oterral — CHANGES_REQUESTED** [review]
I think we can reduce the duplication of code. Call me if my thoughts are not clear :D

File: src/ol/controls/RoutingControl.js (hunk 1 above)
Comment (friedjoff) [reply src/ol/controls/RoutingControl.js]: Good idea! Is done together with JSDoc comment.

File: src/ol/controls/RoutingControl.js (hunk 2 above)
Comment (friedjoff) [reply src/ol/controls/RoutingControl.js:233]: Also really good idea ✅

**Comment by friedjoff** [comment]
@oterral thanks for your suggestions, should be good now.

File: src/ol/layers/RoutingLayer.js [hunk 3]
```diff
@@ -3,13 +3,17 @@ import { Vector as VectorSource } from 'ol/source';
 import { Vector } from 'ol/layer';
 import Layer from './Layer';
 
-const defaultStyleFunction = (feature) => {
+const defaultStyleFunction = (feature, resolution) => {
   const mot = feature.get('mot');
   const viaPointIdx = feature.get('viaPointIdx');
+  const minResolution = feature.get('minResolution');
+  const maxResolution = feature.get('maxResolution');
+  const inRange = resolution <= minResolution && resolution >= maxResolution;
```
Comment (oterral) [inline src/ol/layers/RoutingLayer.js:11]: Can't we just return null in inRange is false ?  instead of setting strokes to null

Comment (friedjoff) [reply src/ol/layers/RoutingLayer.js:11]: Good point, is done.

**Pull request merged**